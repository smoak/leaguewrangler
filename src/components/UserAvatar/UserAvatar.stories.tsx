import { storiesOf } from '@storybook/react';
import * as React from 'react';
import { MockedResponse, MockedProvider } from '@apollo/react-testing';

import getUserAvatar from '../../graphql/queries/getUserAvatar';
import UserAvatar from './UserAvatar';

const mocks: ReadonlyArray<MockedResponse> = [
  {
    request: {
      query: getUserAvatar,
    },
    result: {
      data: {
        currentUser: {
          profilePhotoThumbnailUrl: 'https://picsum.photos/80',
        },
      },
    },
  },
];

storiesOf('UserAvatar', module).add('default', () => (
  <MockedProvider mocks={mocks} addTypename={false}>
    <UserAvatar />
  </MockedProvider>
));
