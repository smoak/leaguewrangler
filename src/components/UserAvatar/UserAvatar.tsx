import { useQuery } from '@apollo/react-hooks';
import React, { FC } from 'react';
import { Avatar, createStyles, withStyles, WithStyles, CircularProgress } from '@material-ui/core';
import AccountCircle from '@material-ui/icons/AccountCircle';

import { GetUserAvatar } from 'graphql/types/GetUserAvatar';
import getUserAvatarQuery from 'graphql/queries/getUserAvatar';

const styles = createStyles({
  avatar: {
    margin: 10,
  },
});

export type UserAvatarProps = WithStyles<typeof styles>;

const UserAvatar: FC<UserAvatarProps> = ({ classes }) => {
  const { loading, data } = useQuery<GetUserAvatar>(getUserAvatarQuery);

  if (loading) {
    return <CircularProgress color="secondary" />;
  }
  if (data && data.currentUser) {
    return <Avatar src={data.currentUser.profilePhotoThumbnailUrl} className={classes.avatar} />;
  }
  return <AccountCircle />;
};

export default withStyles(styles)(UserAvatar);
