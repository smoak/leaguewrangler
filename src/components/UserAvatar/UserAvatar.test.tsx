import { MockedProvider } from '@apollo/client/testing';
import { render, screen, waitForElementToBeRemoved } from '@testing-library/react';

import UserAvatar from './UserAvatar';
import getUserAvatarQuery from '../../graphql/queries/getUserAvatar';

describe('UserAvatar', () => {
  describe('when the component is fetching data', () => {
    const mocks = [
      {
        request: {
          query: getUserAvatarQuery,
        },
        result: {
          data: {
            currentUser: {
              profilePhotoThumbnailUrl: 'https://picsum.photos/40',
            },
          },
        },
      },
    ];

    beforeEach(() => {
      render(
        <MockedProvider mocks={mocks} addTypename={false}>
          <UserAvatar />
        </MockedProvider>
      );
    });

    it('renders as expected', () => {
      expect(screen.getByRole('progressbar')).toBeInTheDocument();
    });
  });

  describe('when the user does not have an avatar', () => {
    const mocks = [
      {
        request: {
          query: getUserAvatarQuery,
        },
        result: {
          data: {},
        },
      },
    ];

    beforeEach(async () => {
      render(
        <MockedProvider mocks={mocks} addTypename={false}>
          <UserAvatar />
        </MockedProvider>
      );

      await waitForElementToBeRemoved(screen.getByRole('progressbar'));
    });

    it('renders as expected', () => {
      expect(screen.getByTestId('avatar-placeholder'));
    });
  });

  describe('when the user does have an avatar', () => {
    const mocks = [
      {
        request: {
          query: getUserAvatarQuery,
        },
        result: {
          data: {
            currentUser: {
              profilePhotoThumbnailUrl: 'https://picsum.photos/40',
            },
          },
        },
      },
    ];

    beforeEach(async () => {
      render(
        <MockedProvider mocks={mocks} addTypename={false}>
          <UserAvatar />
        </MockedProvider>
      );

      await waitForElementToBeRemoved(screen.getByRole('progressbar'));
    });

    it('renders as expected', () => {
      expect(screen.getByRole('img')).toHaveAttribute('src', 'https://picsum.photos/40');
    });
  });
});
