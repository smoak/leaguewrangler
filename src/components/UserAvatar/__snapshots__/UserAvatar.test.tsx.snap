// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAvatar when the component is fetching data renders as expected 1`] = `
<WithStyles(CircularProgress)
  color="secondary"
/>
`;

exports[`UserAvatar when the user does have an avatar renders as expected 1`] = `
<WithStyles(Avatar)
  className="avatar"
  src="foo"
/>
`;

exports[`UserAvatar when the user does not have an avatar renders as expected 1`] = `<pure(AccountCircleIcon) />`;
